src/System.IO/tests/StringWriter/StringWriterTests.cs

```diff
@@ -0,0 +1,369 @@
+            sw.Dispose();
+        }
+
+        [Fact]
+        public static async Task FlushAsyncWorks()
+        {
+            StringBuilder sb = getSb();
+            StringWriter sw = new StringWriter(sb);
+
+            sw.Write(sb.ToString());
+
+            await sw.FlushAsync(); // I think this is a noop in this case
+
+            Assert.Equal(sb.ToString(), sw.GetStringBuilder().ToString());
+        }
+
+        [Fact]
+        public static void MiscWrites()
+        {
+            var sw = new StringWriter();
+            sw.Write('H');
+            sw.Write("ello World!");
+
+            Assert.Equal("Hello World!", sw.ToString());
+        }
+
+        [Fact]
+        public static async Task MiscWritesAsync()
+        {
+            var sw = new StringWriter();
+            await sw.WriteAsync('H');
+            await sw.WriteAsync(new char[] { 'e', 'l', 'l', 'o', ' ' });
+            await sw.WriteAsync("World!");
+
+            Assert.Equal("Hello World!", sw.ToString());
+        }
+
+        [Fact]
+        public static async Task MiscWriteLineAsync()
+        {
+            var sw = new StringWriter();
+            await sw.WriteLineAsync('H');
+            await sw.WriteLineAsync(new char[] { 'e', 'l', 'l', 'o' });
+            await sw.WriteLineAsync("World!");
+
+            Assert.Equal("H\r\nello\r\nWorld!\r\n", sw.ToString());
+        }
+
+        [Fact]
+        public static void GetEncoding()
+        {
+            var sw = new StringWriter();
+            Assert.Equal(Encoding.Unicode.WebName, sw.Encoding.WebName);
+        }
+
+        [Fact]
+        public static void TestWriteMisc()
+        {
+            var sw = new StringWriter();
+
+            sw.Write(true);
+            sw.Write((char)'a');
+            sw.Write(new Decimal(1234.01));
+            sw.Write((double)3452342.01);
+            sw.Write((int)23456);
+            sw.Write((long)long.MinValue);
+            sw.Write((float)1234.50f);
+            sw.Write((UInt32)UInt32.MaxValue);
+            sw.Write((UInt64)UInt64.MaxValue);
+
+            Assert.Equal("Truea1234.013452342.0123456-92233720368547758081234.5429496729518446744073709551615", sw.ToString());
+        }
+
+ [Fact]
+        public static void TestWriteObject()
+        {
+            var sw = new StringWriter();
+            sw.Write(new Object());
+            Assert.Equal("System.Object", sw.ToString());
+        }
+
+        [Fact]
+        public static void TestWriteLineMisc()
+        {
+            var sw = new StringWriter();
+            sw.WriteLine((bool)false);
+            sw.WriteLine((char)'B');
+            sw.WriteLine((int)987);
+            sw.WriteLine((long)875634);
+            sw.WriteLine((Single)1.23457f);
+            sw.WriteLine((UInt32)45634563);
+            sw.WriteLine((UInt64.MaxValue));
+
+            Assert.Equal(@"False
+B
+987
+875634
+1.23457
+45634563
+18446744073709551615
+", sw.ToString());
```

Comment:
Any of these tests (like this one) that are compiled on Windows using verbatim strings that include newlines are going to fail when run on Unix.  When compiled, these verbatim newlines will be encoded as \r\n, but the WriteLines they're being compared against are going to use the newline for that platform, e.g. \n on Unix.  I'd suggest changing these verbatim strings to instead use Environment.NewLine.